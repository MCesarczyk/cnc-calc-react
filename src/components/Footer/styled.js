import styled from "styled-components";

export const FooterWrapper = styled.footer`
  text-align: center;
  margin: 0;
`;

export const FooterContent = styled.p`
  color: ${({ theme }) => theme.color.primary};
  padding: 0 25px;
  font-size: 14px;
`;