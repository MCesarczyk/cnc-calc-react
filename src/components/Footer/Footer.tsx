import { AUTHOR_PAGE_ADDRESS, AUTHOR_PAGE_LABEL, DATE } from 'assets/fixtures/author';
import styled from 'styled-components';

export const Footer = () => (
  <FooterWrapper>
    <FooterContent>
      Copyright &copy; {DATE} by&nbsp;
      <FooterLink href={AUTHOR_PAGE_ADDRESS}>{AUTHOR_PAGE_LABEL}</FooterLink>
    </FooterContent>
  </FooterWrapper>
);

const FooterWrapper = styled.footer`
  text-align: center;
  margin: 0;
`;

const FooterContent = styled.p`
  color: ${({ theme }) => theme.color.primary};
  padding: 0 25px;
  font-size: 14px;
`;

const FooterLink = styled.a`
  color: inherit;
  transition: 0.5s ease-out;

  &:hover {
    color: crimson;
  }
`;
