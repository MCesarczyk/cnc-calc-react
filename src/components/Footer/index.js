import { StyledFooter, FooterContent } from "./styled";

const Footer = ({ children }) => (
  <StyledFooter>
    <FooterContent>
      {children}
    </FooterContent>
  </StyledFooter>
);

export default Footer;