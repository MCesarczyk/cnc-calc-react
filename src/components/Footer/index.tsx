import { AUTHOR_PAGE_ADDRESS, AUTHOR_PAGE_LABEL, DATE } from "../../assets/fixtures/author";
import { FooterWrapper, FooterContent, FooterLink } from "./styled";

const Footer = () => (
  <FooterWrapper>
    <FooterContent>
      Copyright &copy; {DATE} by&nbsp;
      <FooterLink
        href={AUTHOR_PAGE_ADDRESS}
      >
        {AUTHOR_PAGE_LABEL}
      </FooterLink>
    </FooterContent>
  </FooterWrapper>
);

export default Footer;