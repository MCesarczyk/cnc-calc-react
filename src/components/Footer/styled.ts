import styled from "styled-components";

export const FooterWrapper = styled.footer`
  text-align: center;
  margin: 0;
`;

export const FooterContent = styled.p`
  color: ${({ theme }) => theme.color.primary};
  padding: 0 25px;
  font-size: 14px;
`;

export const FooterLink = styled.a`
  color: inherit;
  transition: 0.5s ease-out;

  &:hover {
    color: crimson;
  }
`