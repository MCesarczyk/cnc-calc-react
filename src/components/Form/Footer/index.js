import { useContext } from "react";
import LanguageContext from "../../../features/language/context";
import languages from "../../../assets/fixtures/languages";
import { ButtonsContainer } from "./styled";
import { FormButton } from "../Button";

const FormFooter = () => {
    const { langId } = useContext(LanguageContext);

    return (
        <ButtonsContainer>
            <FormButton type="submit">
                {languages[langId]?.buttonLabel.submit}
            </FormButton>
            <FormButton type="reset">
                {languages[langId]?.buttonLabel.reset}
            </FormButton>
        </ButtonsContainer>
    )
};

export default FormFooter;