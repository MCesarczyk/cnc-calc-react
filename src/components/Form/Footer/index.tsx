import { ButtonsContainer, FooterButton } from "./styled";

interface FormFooterProps {
  buttons: {
    id: number;
    label: string;
    type: "submit" | "reset";
  }[];
}

const FormFooter = ({ buttons }: FormFooterProps) => (
  <ButtonsContainer>
    {buttons.map((button) => (
      <FooterButton key={button.id} type={button.type}>
        {button.label}
      </FooterButton>
    ))}
  </ButtonsContainer>
);

export default FormFooter;
