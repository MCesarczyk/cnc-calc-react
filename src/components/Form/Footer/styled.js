import styled from "styled-components";

export const ButtonsContainer = styled.div`
    max-width: 30rem;
    margin: 0 auto;
    display: flex;
    justify-content: center;
    gap: 0.75rem;

    @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
        flex-direction: column;
        align-items: stretch;
    }
`;