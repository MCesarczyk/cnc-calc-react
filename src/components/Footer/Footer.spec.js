import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { theme } from 'theme';
import { Footer, FooterLink } from './Footer';

test('Footer contains link to author homepage', () => {
  render(
    <ThemeProvider theme={theme}>
      <Footer />
    </ThemeProvider>,
  );

  const content = screen.getByText(/Copyright/i);
  expect(content).toBeInTheDocument();

  const link = screen.getByRole('link');
  expect(link.innerHTML).toBe('Cesarczyk.dev');
  expect(link).toHaveAttribute('href', 'https://home.cesarczyk.dev');
});

test('Footer rendered properly', () => {
  const footerComponent = renderer.create(
    <ThemeProvider theme={theme}>
      <Footer />
    </ThemeProvider>,
  );

  let footer = footerComponent.toJSON();
  expect(footer).toMatchSnapshot();
});

test('Link change style on hover', () => {
  const linkComponent = renderer.create(<FooterLink />);

  let link = linkComponent.toJSON();
  expect(link).toHaveStyleRule('color', 'crimson', {
    modifier: ':hover',
  });
});
