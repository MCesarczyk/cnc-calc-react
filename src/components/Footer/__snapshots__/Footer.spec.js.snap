// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer rendered properly 1`] = `
.c0 {
  text-align: center;
  margin: 0;
}

.c1 {
  color: #1e3246;
  padding: 0 25px;
  font-size: 14px;
}

.c2 {
  color: inherit;
  -webkit-transition: 0.5s ease-out;
  transition: 0.5s ease-out;
}

.c2:hover {
  color: crimson;
}

<footer
  className="c0"
>
  <p
    className="c1"
  >
    Copyright © 
    2022
     by 
    <a
      className="c2"
      href="https://home.cesarczyk.dev"
    >
      Cesarczyk.dev
    </a>
  </p>
</footer>
`;
